### PR TITLE
permission prompt 

### PR DESCRIPTION
**Pull request by ry**

### CODE REVIEW

**Comment by hayd** [comment]
Nice cleanup!

**Comment by hayd** [comment]
Looks like pty is Unix only. Perhaps we should skip these tests on Windows for now?
There may be a workaround...

**Comment by ry** [comment]
@piscisaureus Please review regarding testing a TTY in windows. 
See `tools/permission_prompt_test.py`

File: src/permissions.rs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    if self.allow_write {
+      return Ok(());
+    };
+    // TODO get location (where access occurred)
+    let r = permission_prompt(format!(
+      "🔐 Deno requests write access to \"{}\".",
+      filename
+    ));;
+    if r.is_ok() {
+      self.allow_write = true;
```
Comment (hayd) [inline src/permissions.rs]: I still think this behavior is a little weird: You're asked if it's okay for deno to write foo.txt, you say yes, and deno can write **anything** for the rest of time.

IMO for now we should not use the filename/domain arg until we have finer permissions, either:

- `"🔐 Deno requests write access.` for the rest of execution.
- `"🔐 Deno requests write access to \"{}\"."` but ask every time.

Comment (ry) [reply src/permissions.rs]: I think it's fine for a first pass. This feature is about ergonomics and clicking "yes" a bunch of times is annoying. Maybe we can change the wording to reflect the fact that access is granted for the rest of execution.

**Comment by ry** [comment]
emoji doesn't work in windows - leaving it out.

Bert tested windows manually - there's no easy way to test there. We're going to skip the test in Windows for now.

**Comment by ry** [comment]
@piscisaureus re second patch - importing pty on win fails - so the imports have to be inside the conditional.

**Pull request merged**